### PR TITLE
Add multi-arch-pipeline job

### PR DESCRIPTION
**Pull request by dustymabe**
This is the first attempt at adding a multi-arch pipeline. We'll
call the multi-arch-pipeline job from the main pipeline after QEMU
tests have passed.

This is currently mostly a copy/paste of the original pipeline. In
the future we want to factor out these bits and re-use them between
the main pipeline and the multi-arch-pipeline job.

### CODE REVIEW

**Comment by dustymabe** [comment]
This is a rollup of a few other PRs. I'll remove those commits when those PRs merge. Focus on the last two commits for now.

File: jobs/multi-arch-pipeline.Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+// here; we can look into making them configurable through the template if
+// developers really need to tweak them (note that in the default minimal devel
+// workflow, only the qemu image is built).
+def cosa_memory_request_mb
+if (official) {
+    cosa_memory_request_mb = 6.5 * 1024
+} else {
+    cosa_memory_request_mb = 2.5 * 1024
+}
+cosa_memory_request_mb = cosa_memory_request_mb as Integer
```
Comment (jlebon) [inline jobs/multi-arch-pipeline.Jenkinsfile:111]: Since everything happens on the remote, we don't actually need to allocate a lot of memory here.

**Review by jlebon — APPROVED** [review]
This is OK to start with, but we need to quickly move towards sharing the main business logic.

Until then, I wish there was a way to ensure that any changes done in the primary pipeline are also done in the multi-arch version if necessary. I don't have a good idea for enforcing that though apart from some large comment blocks at the beginning of both files explaining the situation.

File: jobs/multi-arch-pipeline.Jenkinsfile (hunk 1 above)
Comment (dustymabe) [reply jobs/multi-arch-pipeline.Jenkinsfile:111]: yeah i can bump it down +1

**Pull request merged**